content/develop/clients/hiredis/_index.md

```diff
@@ -0,0 +1,133 @@
+---
+categories:
+- docs
+- develop
+- stack
+- oss
+- rs
+- rc
+- oss
+- kubernetes
+- clients
+description: Connect your C application to a Redis database.
+linkTitle: hiredis (C)
+title: hiredis guide (C)
+weight: 9
+---
+
+[`hiredis`](https://github.com/redis/hiredis) is the
+[C language](https://en.wikipedia.org/wiki/C_(programming_language))
+client for Redis.
+The sections below explain how to install `hiredis` and connect your application
+to a Redis database.
+
+`hiredis` requires a running Redis or [Redis Stack]({{< relref "/operate/oss_and_stack/install/install-stack/" >}}) server. See [Getting started]({{< relref "/operate/oss_and_stack/install/" >}}) for Redis installation instructions.
+
+## Build and install
+
+Clone or download the `hiredis` source from the [Github repository](https://github.com/redis/hiredis).
+Then, in a terminal, go into the `hiredis` folder and run the `make` command to build
+the dynamically-loaded library for `hiredis` (this has the name `libhiredis.dylib` on
+MacOS and `libhiredis.so` on Linux). You can copy this library to your
+project folder or run `sudo make install` to install it to `/usr/local/lib`.
+You should also copy the header files `hiredis.h`, `alloc.h`, `read.h`, and
+`sds.h` to your project.
+
+## Connect and test
+
+The code in the example below connects to the server, stores and retrieves
+a string key using [`SET`]({{< relref "/commands/set" >}}) and
+[`GET`]({{< relref "/commands/get" >}}), and then finally closes the
+connection. An explanation of the code follows the example.
+
+```c
+#include <stdio.h>
+
+#include "hiredis.h"
+
+int main() {
+    // The `redisContext` type represents the connection
+    // to the Redis server. Here, we connect to the
+    // default host and port.
+    redisContext *c = redisConnect("127.0.0.1", 6379);
+
+    // Check if the context is null or if a specific
+    // error occurred.
+    if (c == NULL || c->err) {
+        if (c != NULL) {
+            printf("Error: %s\n", c->errstr);
+            // handle error
+        } else {
+            printf("Can't allocate redis context\n");
+        }
+    }
+
+    // Set a string key.
+    redisReply *reply = redisCommand(c, "SET foo bar");
```

Comment:
I'd stick to the format used in the hiredis example where we don't do defensive programming to check the reply object to improve readability:

https://github.com/redis/hiredis/blob/77bcc73ebbc562d8e8173832b710bfbfa4327b13/examples/example.c#L90-L93